### PR TITLE
Feature: Website SDK

### LINKED ISSUE #274: Ability to pass the user information to chatwidget via parameters when initializing a widget.
The js SDK should have the option to update the user information via API

- [ ]  ability to update user details on widget initialize 
- [ ]  ability to label the conversation through the js sdk
- [ ]  trigger chat window from anywhere

### PR DESCRIPTION
**Pull request by pranavrajs**
Fixes #274 

Expose window.$chatwoot object while using SDK

```js
window.chatwootSettings = {
  hideMessageBubble: false,
}
```

### To trigger widget without displaying bubble

```js
window.$chatwoot.toggle()
```

### To set the user in the widget

```js
window.$chatwoot.setUser('identifier', {
  email: 'email@example.com',
  name: 'name',
  avatar_url: '',
})
```

### To set labels on the conversation

```js
window.$chatwoot.addLabel('support-ticket')

window.$chatwoot.removeLabel('support-ticket')
```

### To refresh the session (use this while you logout user from your app)

```js
window.$chatwoot.reset()
```

### CODE REVIEW

File: app/views/widget_tests/index.html.erb [hunk 1]
```diff
@@ -1,6 +1,11 @@
 <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, user-scalable=0" />
 
 <script>
+
+window.chatwootSettings = {
```
Comment (sojan-official) [inline app/views/widget_tests/index.html.erb]: Do we need to set this always?

Comment (pranavrajs) [reply app/views/widget_tests/index.html.erb]: No, defaults will be taken if it is not set

**Comment by vfasoulas** [comment]
what i am doing wrong  ???

(function(d,t) {
    var BASE_URL = "http://chatwoot.hmu.gr";
    var g=d.createElement(t),s=d.getElementsByTagName(t)[0];
    g.src= BASE_URL + "/packs/js/sdk.js";
    s.parentNode.insertBefore(g,s);
    g.onload=function(){

      window.chatwootSDK.run({
        websiteToken: 'PJLMrAsk9sUQYLm9wVhzPPCg',
        baseUrl: BASE_URL
      });

      setTimeout(
        function() {
          console.log("try user set");
          window.$chatwoot.reset();
          window.$chatwoot.setUser('vfasoulas@hmu.gr', {
              email: 'vfasoulas@hmu.gr',
              name: 'fasoulas vasilis',
              avatar_url: '',
          });
          console.log("user set");
        }, 5000);
    }
  })(document,"script");

but the user identified as **Dawn-Cloud-318** without any more infomation, see atachment

image

**Comment by pranavrajs** [comment]
@vfasoulas Moving this to issues. https://github.com/chatwoot/chatwoot/issues/742

**Pull request merged**